### ISSUE TITLE
Async Multiselect, options dropdown hides every time an option selected

### ISSUE THREAD

**Issue by dspoonia7** [JedWatson/react-select#1542]
In Async Multiselect (Select.Async), if multi-select is true then options dropdown hides every time an option selected. To see the dropdown I have to search/type. I want to keep dropdown opened instead of hiding it every time after selecting an option like happens in normal Multiselect. 

A gif is attached showing issue

![react-select-async-multi](https://cloud.githubusercontent.com/assets/8257269/22774318/986d25e0-eecc-11e6-82a0-83c8ed4d2ceb.gif)

**Comment by cordoval** [comment]
i desire to have the same effect of closing the options dropdown automatically but on the first case of basic multiselect. Can anyone hint me how to do this please?

**Comment by dspoonia7** [comment]
@cordoval try setting `autoBlur={true}`. See #1246

**Comment by cordoval** [comment]
thanks!

**Comment by sgoran** [comment]
Have the same problem with async and multiselect

**Comment by literatura** [comment]
I have the same problem with async and multiple. The option `autoBlur={true}` does not help. 
Му code:
`<Select.Async
                className='selectFilter'
                name='form-field-name'
                value={curValues}
                multi={true}
                autoload={false}
                autoBlur={true}
                backspaceRemoves={false}
                menuContainerStyle={{ zIndex: 5 }}
                loadOptions={this.getOptions}
                onChange={this.handleChangeSelect.bind(this, someKey)}
                cache={false} />`

**Comment by dherran** [comment]
This was suggested and pull requested a long time ago #882

**Comment by w3-3w** [comment]
Since Jed approved changes in #1289 , while still someone need to keep the options after selection, I think it is better to add a configurable prop to Async, defining whether options should be cleared after selection on Async when `multi` is true.

**Comment by RafaelSalguero** [comment]
I have the same problem

**Comment by begincalendar** [comment]
@dherran #882 doesn't seem to be related to this, as far as I see.
The functionality that #882 strives for seems to already existed in the latest version, so I'm not sure why that pull request is still open.

#1561 appears to be a duplicate of this issue.

**Comment by tridip1931** [comment]
@dspoonia7 Did you have the solution of this? I have the same issue. I am using Async

**Comment by begincalendar** [comment]
@tridip1931 A workaround would be to extend the `Async` class and override the `clearOptions()` method, such that it does nothing.

**Comment by dspoonia7** [comment]
Hey @tridip1931, Yes I was able to get it working (It's been a while so not actually remember I got this). This solution worked for me.
```
  getOption = (query, callback) => {
      // In http call response
      const options = [{ value: <val>, label: <val> }, ...]
     callback(null, {
        options: options
      })
  }
```
```
   const selectedValues = [{ value: <val>, label: <val>, name: <val> }, ...]

  <Select.Async
      ref='<ref>'
      value={selectedValues}
      loadOptions={this.getOption}
      autosize={true}
      multi={true}
      clearable={false}
      cache={false}
      onChange={(item) => this.handleOnChange(item)}
      onFocus={this.handleOnFocus}
    />

```
Hope this is helpful :)

**Comment by tridip1931** [comment]
Thanks @dspoonia7 @begincalendar I am quite new to `React` so forgive my dumb questions.
I am using `AsyncCreatable` to generate the muliselect option.

@begincalendar Can you elaborate where to use clearOptions()?

@dspoonia7 What different from the default component you did that made it work?

Thanks 🙇

**Comment by begincalendar** [comment]
@tridip1931 Take a look at how `clearOptions()` is used [here](https://github.com/JedWatson/react-select/blob/master/src/Async.js#L193-L195) and its implementation [here](https://github.com/JedWatson/react-select/blob/master/src/Async.js#L78-L80).

So if you do something like this:
```
class MyAsync extends Select.Async {
  clearOptions() {
  }
}
```

You are working around the issue by making `clearOptions()` do nothing, instead of the undesirable behaviour which is the focus of this issue.

**Comment by JedWatson** [comment]
I've changed the way this whole feature works (the changes will be published as rc.9 or final) so you can control this behaviour now. The menu will always close by default for both single and multi-select, but you can change it with the `closeOnSelect={false}` prop. To keep suggestions from the Async component after selecting an option, also set `onSelectResetsInput={false}`

**Comment by tommyalvarez** [comment]
@JedWatson is this implemented in "react-select": "^1.0.0-rc.3", ? Because that option is not working for me

**Comment by manjuprem12** [comment]
Reactjs TypeError: Unable to get property 'string' of undefined or null reference.
i am getting the above error while implementing Async options. I was trying with react class component not using createclass
reference :  https://reactjsexample.com/a-fork-of-react-select-with-support-for-option-groups/ 
Please give me some solution.
https://stackoverflow.com/questions/59420913/reactjs-typeerror-unable-to-get-property-string-of-undefined-or-null-referenc?noredirect=1#comment105028760_59420913
